controller.py

```diff
@@ -238,7 +303,7 @@ def add_with_metadata(self, collection_details):
         entries = feed.get("entries", [])
 
         if not client.data_source:
-            client.data_source = DataSource.lookup(self._db, client.key, autocreate=True)
+            client.data_source = DataSource.lookup(self._db, client.url, autocreate=True)
```

Comment:
This will mean that the DataSource is tied to the overarching server from which the library edit came -- not to any particular library on that server.

Reply:
How does this situation differ from https://github.com/NYPL-Simplified/metadata_wrangler/pull/123/files#diff-7a315c65deca2c033ce485bfa9de4f4aR225, where we use client.key?

Reply:
Upon further reflection I think I understand why we use client.url here, but now I think we should also use client.url in the other place as well. So either we need another change or I'm confused.

Reply:
It's not different from the previous change -- it's just that client.key doesn't exist anymore in this branch. Either way, changes would be server-specific, not library-specific.